lib/mayaUsd/ufe/MayaUIInfoHandler.cpp

```diff
@@ -0,0 +1,144 @@
+//
+// Copyright 2022 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "MayaUIInfoHandler.h"
+
+#include <mayaUsd/fileio/primUpdaterManager.h>
+#include <mayaUsd/ufe/Global.h>
+#include <mayaUsd/ufe/UsdSceneItem.h>
+#include <mayaUsd/utils/util.h>
+
+#include <pxr/base/tf/diagnostic.h>
+
+#include <ufe/hierarchy.h>
+#include <ufe/pathString.h>
+
+// Needed for Tf diagnostic macros.
+PXR_NAMESPACE_USING_DIRECTIVE
+
+namespace {
+
+// Find a pulled Maya node's USD pulled ancestor by iterating up the Maya
+// path.  Iteration stops when a Maya node with pull information has been
+// found, which can be for the initial path itself, when the Maya node is
+// the root of the pulled sub-hierarchy.
+//
+// Note that if it exists, the USD pulled ancestor prim is inactive, which is
+// done to avoid having that prim and its sub-hierarchy exist in the scene as
+// stale USD duplicates of Maya pulled nodes.  If the USD pulled ancestor does
+// not exist, the argument Maya node is orphaned.
+MayaUsd::ufe::UsdSceneItem::Ptr pulledUsdAncestorItem(const Ufe::SceneItem::Ptr& mayaItem)
+{
+    // This function requires a Maya item to compute its USD ancestor.
+    TF_AXIOM(mayaItem->runTimeId() == MayaUsd::ufe::getMayaRunTimeId());
```

Comment:
Isn't that pretty harsh to abort on this condition (and the one below)? Why not just return nullptr?